src/utils/log.c

```diff
@@ -160,6 +162,11 @@ void log_msg(log_level level, const char *fmt, ...) {
         va_start(args, fmt);
         vfprintf(target->fp, fmt, args);
         va_end(args);
+        if(level == LOG_ERROR) {
+            va_start(args, fmt);
```

Comment:
Calling `va_start` twice on the same argument is UB. You need to use `va_copy`

Reply:
I thought va_start() reinitialized the list safely? Do you have a link to anything that discusses this? I'm having trouble finding anything.

Reply:
The linux man page for va_start says: `Multiple traversals of the list, each bracketed by va_start() and va_end() are possible. va_end() may be a macro or a function.`

Reply:
Seems to have been changed recently then.

Reply:
Happy to change it if you can find a reference that shows it might be UB. UBsan doesn't seem to flag it either?

Reply:
cppcheck at least used to.

Reply:
I installed cppcheck and ran it on this file (and the whole project). It did find some other stuff, but nothing about va_args here.

Reply:
I remember having to do va_copy at some point too, though it's been a while. Probably something that was changed. Or possibly mandela effect.